src/compiler/crystal/tools/doc/crystal_pseudo_methods.cr

```diff
@@ -0,0 +1,149 @@
+class Crystal::Doc::Generator
+  def insert_pseudo_methods
+    object = @program.object
+
+    object.add_def declare_pseudo_method("!",
+      return_type: Path.new(["Bool"]),
+      doc: <<-DOC)
+        Returns the boolean negation of `self`.
+
+        ```
+        !true   # => false
+        !false  # => true
+        !nil    # => true
+        !1      # => false
+        !"foo"  # => false
+        ```
+
+        This method is a unary operator and usually written in prefix notation
+        (`!foo`) but it can also be written as a regular method call (`foo.!`).
+        DOC
+
+    object.add_def declare_pseudo_method("is_a?",
+      args: [
+        Arg.new("type", restriction: Path.new(["Class"])),
+      ],
+      return_type: Path.new(["Bool"]),
+      doc: <<-DOC)
+        Returns `true` if  inherits or includes *type*.
```

Comment:
```suggestion
        Returns `true` if  `self` inherits or includes *type*.
```

Reply:
Note: there's 2 spaces after the `if`, `self` can go right in the middle